nctl/nightly-tests: several script fixes for parsing blocks

alsrdn opened the pull request:
Several script changes to support some NCTL functions with feat-2.0.

alsrdn committed:
nctl/nightly-tests: several script fixes for parsing blocks

Signed-off-by: Alexandru Sardan <alexandru@casperlabs.io>

jacek-casper commented on utils/nctl/sh/scenarios/common/itst.sh:
```diff
@@ -296,7 +296,12 @@ function get_switch_block() {
 
     if [ "$WALKBACK" -gt 0 ]; then
-        BLOCK=$(echo "$JSON_OUT" | jq '.result.block')
+        VERSIONED_BLOCK="$(echo "$JSON_OUT" | jq '.result.block_with_signatures.block')"
+        if jq -e '.Version2' <<< "$VERSIONED_BLOCK" >/dev/null; then
+            BLOCK="$(echo "$VERSIONED_BLOCK" | jq '.Version2')"
+        else
+            BLOCK="$(echo "$VERSIONED_BLOCK" | jq '.Version1')"
+        fi
```
might be worth pulling out to a function, we repeat this a lot now

rafal-ch replied:
yes, that's potentially worth doing, but I'm also not pushing too much because we'll need to rework this stuff in the future when we'll be working on test coverage for all legacy data types

rafal-ch reviewed (approved):
LGTM :+1:

alsrdn commented:
bors r+

fizyk20 commented on utils/nctl/sh/scenarios/configs/swap_validator_set.config.toml.override:
```diff
@@ -0,0 +1,2 @@
+[binary_port_server]
+allow_request_get_all_values = true
```
Wouldn't it be easier to just set it in `resources/local/config.toml`? Or is there a reason not to do that?

alsrdn replied:
Yes, probably easier to add it there. Although in `local/config.toml` it's disabled specifically by default intentionally.

fizyk20 replied:
I mean, I understand why it's disabled in the production config, but local config is only used for tests, isn't it?

alsrdn replied:
@rafal-ch any reason why it was disabled in the local config intentionally?

rafal-ch replied:
None that I know of. Maybe just for the sake of keeping these two configs as much in sync as possible.

alsrdn replied:
@fizyk20 set it to false in the local config as suggested here: https://github.com/casper-network/casper-node/pull/4603/commits/b2763eca4b6fb25a8bef2ddf9231f348f5c83509

alsrdn committed:
Merge branch 'feat-2.0' into fix-nightlies

alsrdn committed:
resources/local config: set allow_request_get_all_values true

Set allow_request_get_all_values to true to allow inspection of auction
data in NCTL tests.

Signed-off-by: Alexandru Sardan <alexandru@casperlabs.io>

alsrdn commented:
bors merge

alsrdn commented:
bors merge

alsrdn committed:
nctl: fix `event_stream.sh` nightly test scenario.

The `TransactionAccepted` event now replaces the old `DeployAccepted`
event.

Signed-off-by: Alexandru Sardan <alexandru@casperlabs.io>

alsrdn committed:
Merge branch 'feat-2.0' into fix-nightlies

alsrdn commented:
bors single on

alsrdn commented:
bors merge p=1
